server/datastore/mysql/software.go

```diff
@@ -2872,243 +2278,753 @@ last_vpp_install AS (
+			LEFT JOIN
+				software_installers ON software_titles.id = software_installers.title_id 
+				AND software_installers.global_or_team_id = :global_or_team_id
+			INNER JOIN 
+				software ON software_titles.id = software.title_id
+			` + vulnerableJoin + `
 			WHERE
-				sc.software_id IN (?)
-			ORDER BY
-				software_id, cve
-	`
-			type softwareCVE struct {
-				SoftwareID uint   `db:"software_id"`
-				CVE        string `db:"cve"`
-			}
-			var softwareCVEs []softwareCVE
-			stmt, args, err = sqlx.In(cveStmt, softwareIDs)
+				software.id IN (?) 
+				AND true
+			` + softwareOnlySelfServiceClause + `
+			-- GROUP by for software
+			%s
+			`
+
+			softwareTitleStatment, softwareTitleArgs, err := sqlx.In(softwareTitleStatment, softwareIds)
 			if err != nil {
-				return nil, nil, ctxerr.Wrap(ctx, err, "building query args to list cves")
+				return nil, nil, ctxerr.Wrap(ctx, err, "expand IN query for list host software")
 			}
-			if err := sqlx.SelectContext(ctx, ds.reader(ctx), &softwareCVEs, stmt, args...); err != nil {
-				return nil, nil, ctxerr.Wrap(ctx, err, "list software cves")
+			softwareTitleStatment, softwareTitleArgsNamedArgs, err := sqlx.Named(softwareTitleStatment, namedArgs)
+			if err != nil {
+				return nil, nil, ctxerr.Wrap(ctx, err, "build named query for list host software")
 			}
-
-			// store the CVEs with the proper software entry
-			for _, cve := range softwareCVEs {
-				ver := bySoftwareID[cve.SoftwareID]
-				ver.Vulnerabilities = append(ver.Vulnerabilities, cve.CVE)
+			softwareTitleStatment = strings.Replace(softwareTitleStatment, "AND true", matchClause, -1)
+			args = append(args, softwareTitleArgsNamedArgs...)
+			args = append(args, softwareTitleArgs...)
+			if len(matchArgs) > 0 {
+				args = append(args, matchArgs...)
 			}
-
-			const pathsStmt = `
-			SELECT
-				hsip.software_id,
-				hsip.installed_path,
-				hsip.team_identifier
+			stmt = stmt + softwareTitleStatment
+		}
+		if len(vppAdamIDs) > 0 {
+			vppAdamStatment = `
+			UNION
+			-- SELECT for vpp apps
+			%s
 			FROM
-				host_software_installed_paths hsip
+				software_titles
+			LEFT JOIN
+				software_installers ON software_titles.id = software_installers.title_id
+				AND software_installers.global_or_team_id = :global_or_team_id
+			INNER JOIN
+				vpp_apps ON software_titles.id = vpp_apps.title_id AND vpp_apps.platform = :host_platform
+			INNER JOIN
+				vpp_apps_teams ON vpp_apps.adam_id = vpp_apps_teams.adam_id AND vpp_apps.platform = vpp_apps_teams.platform AND vpp_apps_teams.global_or_team_id = :global_or_team_id
 			WHERE
-				hsip.host_id = ? AND
-				hsip.software_id IN (?)
-			ORDER BY
-				software_id, installed_path
-	`
-			type installedPath struct {
-				SoftwareID     uint   `db:"software_id"`
-				InstalledPath  string `db:"installed_path"`
-				TeamIdentifier string `db:"team_identifier"`
+				vpp_apps_teams.adam_id IN (?) 
+				AND true
+			` + vppOnlySelfServiceClause + `
+			-- GROUP BY for vpp apps
+			%s
```

Comment:
Do we need this second query just for VPP apps? Can we merge this with the query above and avoid the union?